debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSnapshotChangeEventSource.java

```diff
@@ -394,7 +400,27 @@ protected void complete(SnapshotContext<MySqlOffsetContext> snapshotContext) {
      */
     @Override
     protected Optional<String> getSnapshotSelect(RelationalSnapshotContext<MySqlOffsetContext> snapshotContext, TableId tableId) {
-        return Optional.of(String.format("SELECT * FROM `%s`.`%s`", tableId.catalog(), tableId.table()));
+        String snapshotSelectColumns = getSnapshotSelectColumns(tableId);
+        return Optional.of(String.format("SELECT %s FROM `%s`.`%s`", snapshotSelectColumns, tableId.catalog(), tableId.table()));
+    }
+
+    @Override
+    protected String getSnapshotSelectColumns(TableId tableId) {
```

Comment:
We can take this method in each connector one step further and achieve what Gunnar suggested.  In each connector we effectively do the following:

1. Get the `Table` from the schema.
2. Make a call to `prepareSnapshotSelectColumns(Table, TableId)`
3. Iterate the return column names and format them.

I think the `prepareSnapshotSelectColumns` method can be adjusted to accept a lambda or functional interface that does (3).  It would accept the column list as an input argument, and it would be responsible for doing all the necessary formatting and filtering steps needed by the connector implementation.  This would then mean the common method would be responsible for doing (1) in conjunction with what it already does so each of the individual connector implementations would look like:

```
protected String getSnapshotSelectColumns(TableId tableId) {
  return prepareSnapshotSelectSelectColumns(tableId, (columns) -> {
    return columns.stream().filter( ... ).map( ... ).collect(Collectors.joining(","));
  });
}
```

Does that make sense?